From 8a2f5d946319fbf454313e5184e329929c2b00e9 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Sat, 18 Jan 2014 08:38:13 +0000
Subject: [PATCH 089/351] switch: arizona: Add configurable delay before
 MICBIAS ramp

Allow users to add a configurable delay between detection of a jack and
ramping the MICBIAS to being microphone detection.

Change-Id: I8db53ade7efabb90f4c86028b84daa3026eecc50
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 429af47..2c07155 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -28,6 +28,7 @@
 #include <linux/switch.h>
 #include <linux/of.h>
 #include <linux/device.h>
+#include <linux/delay.h>
 
 #include <sound/soc.h>
 
@@ -1046,6 +1047,9 @@ static irqreturn_t arizona_jackdet(int irq, void *data)
 			info->mic = false;
 			info->jack_flips = 0;
 
+			if (arizona->pdata.init_mic_delay)
+				msleep(arizona->pdata.init_mic_delay);
+
 			arizona_start_mic(info);
 		} else {
 			schedule_delayed_work(&info->hpdet_work,
@@ -1161,6 +1165,9 @@ static int arizona_extcon_get_pdata(struct arizona *arizona)
 
 	arizona_of_read_u32(arizona, "wlf,gpsw", false, &pdata->gpsw);
 
+	arizona_of_read_u32(arizona, "wlf,init-mic-delay", false,
+			    &pdata->init_mic_delay);
+
 	return 0;
 }
 
-- 
1.7.9.5

